src/sagemaker/tensorflow/estimator.py

```diff
@@ -363,17 +409,41 @@ def hyperparameters(self):
         hyperparameters = super(TensorFlow, self).hyperparameters()
 
         if not self.checkpoint_path:
-            local_code = get_config_value('local.local_code', self.sagemaker_session.config)
-            if self.sagemaker_session.local_mode and local_code:
-                self.checkpoint_path = '/opt/ml/shared/checkpoints'
-            else:
-                self.checkpoint_path = os.path.join(self.output_path,
-                                                    self._current_job_name, 'checkpoints')
-
-        additional_hyperparameters = {'checkpoint_path': self.checkpoint_path,
-                                      'training_steps': self.training_steps,
-                                      'evaluation_steps': self.evaluation_steps,
-                                      'sagemaker_requirements': self.requirements_file}
+            self.checkpoint_path = self._default_s3_path('checkpoints')
+
+        if self.script_mode_enabled():
+            if not self.model_dir:
+                self.model_dir = self._default_s3_path('model')
+            additional_hyperparameters = {'model_dir': self.model_dir}
+            if self.distributions:
```

Comment:
Let's make self.distributions default empty instead of none so you do not need to check it.